wagtail/snippets/views/snippets.py

```diff
@@ -868,6 +868,8 @@ def edit_view(self):
             template_name=self.get_edit_template(),
             header_icon=self.icon,
             permission_policy=self.permission_policy,
+            panel=self._edit_handler,
+            form_class=self.get_form_class(),
```

Comment:
Not that it makes a difference, but I forgot to add `for_update=True` here (and in the `revisions_revert` view)